### PR TITLE
Aggregations in search state

### PR DESCRIPTION
**Pull request by LHBruneton-C2C**
Add aggregations to search state (request and response)
Add aggregations to query payload
Read aggregations from response
Display facets in app

### CODE REVIEW

File: libs/search/src/lib/facets/facets-container/facets-container.component.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import { Component, Input, OnInit } from '@angular/core'
+import { select, Store } from '@ngrx/store'
+import { map } from 'rxjs/operators'
+import { SearchState } from '../../state/reducer'
+import { getSearchResultsAggregations } from '../../state/selectors'
+import { FacetsService } from '../facets.service'
```
Comment (fxprunayre) [inline libs/search/src/lib/facets/facets-container/facets-container.component.ts:6]: Would it make sense to use Aggregations or Facets everywhere for consistency ?

If yes, I would vote for Aggregation. 

Components in https://github.com/geonetwork/geonetwork-ui/pull/38 could be Aggs > Agg > AggBuckets > AggBucket

Comment (fgravin) [reply libs/search/src/lib/facets/facets-container/facets-container.component.ts:6]: `aggregations` is a way elastic aggregates the data, `facet` is a UI representation, other components like `metrics` can use aggregations: I would keep the distanciation.

File: libs/ui/src/lib/facets/facet-block/facet-block.component.ts [hunk 2]
```diff
@@ -53,3 +53,13 @@ export class FacetBlockComponent implements OnInit {
     return this.model.more
   }
 }
+
+@Component({ selector: 'ui-facet-block', template: '' })
```
Comment (fgravin) [inline libs/ui/src/lib/facets/facet-block/facet-block.component.ts:57]: I wouldn't put that here, why do you need this stub byt the way ?

File: libs/search/src/lib/state/reducer.ts [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
+import { RecordSummary, SearchFilters } from '@lib/common'
 import * as fromActions from './actions'
 
 export const SEARCH_FEATURE_KEY = 'searchState'
 
 export interface SearchState {
-  params: SearchParams
-  sortBy?: string
-  results: RecordSummary[]
+  requestParams: {
```
Comment (fgravin) [inline libs/search/src/lib/state/reducer.ts:7]: I would change the structure of the state following our discussions.

**Review by fgravin — CHANGES_REQUESTED** [review]
Good so far, the smart logic is well connected to the state.
Some refactoring of the state structure is expected though.
Thanks

**Comment by fgravin** [comment]
@LHBruneton-C2C I don't see the translation of `facets.block.title.tag`

**Comment by LHBruneton-C2C** [comment]
> @LHBruneton-C2C I don't see the translation of `facets.block.title.tag`

Neither do I, even though it was working OK before the rebase with the last commits on master. I will look into it.

File: libs/ui/src/lib/facets/facet-block/facet-block.component.ts (hunk 2 above)
Comment (LHBruneton-C2C) [reply libs/ui/src/lib/facets/facet-block/facet-block.component.ts:57]: I use it to test the facet-list component without additional schemas. Should we just use the NO_ERRORS_SCHEMA everywhere ?

Comment (fgravin) [reply libs/ui/src/lib/facets/facet-block/facet-block.component.ts:57]: It depends what you do want to test. But usually, when testing a component, we just test the inner behavior of the component, and don't dive deeper in the components tree.

**Comment by fgravin** [comment]
> Neither do I, even though it was working OK before the rebase with the last commits on master. I will look into it.

Ok, also I don't see the facet item text color change on hover, I just see the checkbox the slighlty get darker.

**Comment by LHBruneton-C2C** [comment]
> I don't see the facet item text color change on hover, I just see the checkbox the slighlty get darker.
Don't know what's happening here. On my machine, no diff with the branch, npm run storybook, I see the color change on hover. I'm using chromium. What platform are you trying on ? Is this the result of the storybook compiled by the CI ?

**Comment by LHBruneton-C2C** [comment]
> No, I'm trying on `app-search`

Well, we have a "It works on my machine" issue... ¯\_(ツ)_/¯
Is there a continuous deployment on this project ?

**Comment by fgravin** [comment]
> Is there a continuous deployment on this project ?

storybook but disabled for now ..
I will try to find time to fix it and also deploy the applications as well

**Comment by LHBruneton-C2C** [comment]
> > Is there a continuous deployment on this project ?
> 
> storybook but disabled for now ..
> I will try to find time to fix it and also deploy the applications as well

Thanks !

File: libs/ui/src/lib/facets/facet-block/facet-block.component.ts (hunk 2 above)
Comment (LHBruneton-C2C) [reply libs/ui/src/lib/facets/facet-block/facet-block.component.ts:57]: As explained here [https://angular.io/guide/testing-components-scenarios#nested-component-tests](url), the stub is only here to test the nested component interface, nothing deeper.

**Review by fgravin — APPROVED** [review]
Excellent.
Thanks for the refactoring and pluggin everything together.
Now we need to perform a search on facet item click :)

**Pull request merged**